scripts/ci/validatebuildlog.ps1

```diff
@@ -0,0 +1,75 @@
+<#
+.SYNOPSIS
+    Validates a build log to check for the presence of non-build related warnings
+    that may negatively affect the quality of the build.
+.DESCRIPTION
+    This script intends to prevent repeats of past issues that have come up
+    (for example, reused GUIDs) which don't get caught in the build itself
+    because they are not build errors.
+
+    Returns 0 if there are no errors, non-zero if there are.
+.PARAMETER LogFile
+    The log file to validate
+.EXAMPLE
+    .\validatebuildlog.ps1 -LogFile c:\path\to\log.txt
+#>
+param(
+    [string]$LogFile
+)
+
+function CheckDuplicateGuids(
+    [string[]]$LogFileContent,
+    [int]$LineNumber
+) {
+    <#
+    .SYNOPSIS
+        Checks if the given log file has a conflict guid at the specified line number.
+        Outputs to console if such an error exists.
+        Returns true if there is a duplicate guid.
+    #>
+    if ($LogFileContent[$LineNumber] -match "GUID \[[a-g0-9]{32}?\] for asset '.*' conflicts with") {
+        for ($i = $LineNumber; $i -lt $LogFileContent.Length; $i++) {
+            if ($LogFileContent[$i] -eq "Assigning a new guid.") {
+                Write-Host "Found duplicated GUID, Unity will non-deterministically use one of them, please manually"
+                    "regenerate the intended one by deleting the .meta file and re-opening the Unity editor locally"
```

Comment:
![image](https://user-images.githubusercontent.com/3580640/62574456-6485e580-b84d-11e9-8e98-ea4f4af69aa5.png)
Doesn't look like this line prints

Reply:
Oh my